xbmc/filesystem/MusicDatabaseDirectory.cpp

```diff
@@ -114,9 +114,18 @@ bool CMusicDatabaseDirectory::IsArtistDir(const std::string& strDirectory)
 
 bool CMusicDatabaseDirectory::HasAlbumInfo(const std::string& strDirectory)
 {
-  NODE_TYPE node=GetDirectoryType(strDirectory);
-  return (node!=NODE_TYPE_OVERVIEW && node!=NODE_TYPE_TOP100 &&
-          node!=NODE_TYPE_GENRE && node!=NODE_TYPE_ARTIST && node!=NODE_TYPE_YEAR);
+  NODE_TYPE node = GetDirectoryType(strDirectory);
+  if (node == NODE_TYPE_ALBUM) return true;
+  if (node == NODE_TYPE_ALBUM_RECENTLY_ADDED) return true;
+  if (node == NODE_TYPE_ALBUM_RECENTLY_ADDED_SONGS) return true;
+  if (node == NODE_TYPE_ALBUM_RECENTLY_PLAYED) return true;
+  if (node == NODE_TYPE_ALBUM_RECENTLY_PLAYED_SONGS) return true;
+  if (node == NODE_TYPE_ALBUM_TOP100) return true;
+  if (node == NODE_TYPE_ALBUM_TOP100_SONGS) return true;
+  if (node == NODE_TYPE_ALBUM_COMPILATIONS) return true;
+  if (node == NODE_TYPE_ALBUM_COMPILATIONS_SONGS) return true;
+  if (node == NODE_TYPE_YEAR_ALBUM) return true;
+  return false;
 }
```

Comment:
Thanks @Razzeee @AlwinEsch for code suggestions/corrections. 

I am actually wondering if `HasAlbumInfo` should be deprecated anyway. It is used in GUIWindowMusicNav to check if we are showing albums so that the "Query info for all" context menu item can be added, and in GUIWindowMusicBase to decide to scrape artists or albums. Surely it would be better just to check `m_vecItems->GetContent()` rather than GetPath, create a CMusicDatabaseDirectory from the path, create a node from the dir and finally check the resulting node is one of the album related ones?